src/components/JobsPanelDataInputs/JobsPanelDataInputs.js

```diff
@@ -56,78 +63,75 @@ const JobsPanelDataInputs = ({
       }
     }
   }, [
-    pathScheme,
     inputsState.newInput.path.project.length,
     inputsState.projects.length,
     match.params.projectName,
-    projectStore.projects
+    projectStore.projects,
+    inputsState.newInput.path.pathType,
+    inputsState.selectedDataInput.data.path.value,
+    inputsState.selectedDataInput
   ])
 
   useEffect(() => {
     if (
       inputsState.artifacts.length === 0 &&
-      inputsState.newInputProjectPathEntered
+      inputsState.inputProjectPathEntered
     ) {
-      fetchArtifacts({ project: inputsState.newInput.path.project }).then(
-        artifacts => {
-          const artifactsList = artifacts
-            .map(artifact => ({
-              label: artifact.link_iteration
-                ? artifact.link_iteration.db_key
-                : artifact.key ?? '',
-              id: artifact.link_iteration
-                ? artifact.link_iteration.db_key
-                : artifact.key ?? ''
-            }))
-            .filter(artifact => artifact.label !== '')
-
-          inputsDispatch({
-            type: inputsActions.SET_ARTIFACTS,
-            payload: artifactsList
-          })
-        }
-      )
+      fetchArtifacts({
+        project:
+          inputsState.newInput.path.project ||
+          inputsState.selectedDataInput.data.path.value.split('/')[0]
+      }).then(artifacts => {
+        const artifactsList = artifacts
+          .map(artifact => ({
+            label: artifact.link_iteration
+              ? artifact.link_iteration.db_key
+              : artifact.key ?? '',
+            id: artifact.link_iteration
+              ? artifact.link_iteration.db_key
+              : artifact.key ?? ''
+          }))
```

Comment:
You can reduce duplication using a simple variable:
```suggestion
          .map(artifact => {
            const key = artifact.link_iteration
              ? artifact.link_iteration.db_key
              : artifact.key ?? ''
            return {
              label: key,
              id: key
            }
          })
```